src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -531,5 +542,707 @@ public void Add(NamespaceOrTypeSymbol symbol)
+                }
+            }
+
+            Debug.Assert(false);
+            return ImmutableDictionary<string, AliasAndUsingDirective>.Empty;
+        }
+
+        public ImmutableArray<NamespaceOrTypeAndUsingDirective> GetUsingNamespacesOrTypes(CSharpSyntaxNode declarationSyntax, ConsList<TypeSymbol>? basesBeingResolved)
+        {
+            switch (declarationSyntax)
+            {
+                case CompilationUnitSyntax compilationUnit:
+                    if (!compilationUnit.Usings.Any())
+                    {
+                        return ImmutableArray<NamespaceOrTypeAndUsingDirective>.Empty;
+                    }
+                    break;
+
+                case NamespaceDeclarationSyntax namespaceDecl:
+                    if (!namespaceDecl.Usings.Any())
+                    {
+                        return ImmutableArray<NamespaceOrTypeAndUsingDirective>.Empty;
+                    }
+                    break;
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(declarationSyntax);
+            }
+
+            foreach (var declaration in _mergedDeclaration.Declarations)
+            {
+                var declarationSyntaxRef = declaration.SyntaxReference;
+                if (declarationSyntaxRef.SyntaxTree != declarationSyntax.SyntaxTree)
+                {
+                    continue;
+                }
+
+                if (declarationSyntaxRef.GetSyntax() == declarationSyntax)
+                {
+                    return _aliasesAndUsings[declaration].GetUsingNamespacesOrTypes(this, declarationSyntax, basesBeingResolved);
+                }
+            }
+
+            Debug.Assert(false);
+            return ImmutableArray<NamespaceOrTypeAndUsingDirective>.Empty;
+        }
+
+        private class AliasesAndUsings
+        {
+            private ExternAliasesAndDiagnostics? _lazyExternAliases;
+            private UsingsAndDiagnostics? _lazyUsings;
+            private Imports? _lazyImports;
+
+            /// <summary>
+            /// Completion state that tracks whether validation was done/not done/currently in process. 
+            /// </summary>
+            private SymbolCompletionState _state;
+
+            internal ImmutableArray<AliasAndExternAliasDirective> GetExternAliases(SourceNamespaceSymbol declaringSymbol, CSharpSyntaxNode declarationSyntax)
+            {
+                return GetExternAliasesAndDiagnostics(declaringSymbol, declarationSyntax).ExternAliases;
+            }
+
+            private ExternAliasesAndDiagnostics GetExternAliasesAndDiagnostics(SourceNamespaceSymbol declaringSymbol, CSharpSyntaxNode declarationSyntax)
+            {
+                if (_lazyExternAliases is null)
+                {
+                    SyntaxList<ExternAliasDirectiveSyntax> externAliasDirectives;
+                    switch (declarationSyntax)
```

Comment:
nit: could be compacted into a switch expression (also in `GetUsingsAndDiagnostics`)

Reply:
> could be compacted into a switch expression

In general, I find switch expressions harder to read and follow.

---
In reply to: [588799753](https://github.com/dotnet/roslyn/pull/51676#discussion_r588799753) [](ancestors = 588799753)